<?xml version='1.0' encoding='UTF-8'?>
<root>
 <tabbed_widget name="Main Window" parent="main_window">
  <Tab containers="1" tab_name="pos">
   <Container>
    <DockSplitter count="2" orientation="-" sizes="0.500564;0.499436">
     <DockSplitter count="3" orientation="|" sizes="0.334192;0.331616;0.334192">
      <DockArea name="...">
       <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
        <range right="12.678383" left="11.921057" top="0.057138" bottom="-0.095979"/>
        <limitY/>
        <curve color="#1f77b4" name="/iiwa_front/bspline_ff_joint_trajectory_controller/state/actual/positions.0"/>
        <curve color="#17becf" name="/iiwa_front/bspline_ff_joint_trajectory_controller/state/desired/positions.0"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
        <range right="12.678383" left="11.921057" top="1.133868" bottom="0.796736"/>
        <limitY/>
        <curve color="#d62728" name="/iiwa_front/bspline_ff_joint_trajectory_controller/state/actual/positions.1"/>
        <curve color="#bcbd22" name="/iiwa_front/bspline_ff_joint_trajectory_controller/state/desired/positions.1"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
        <range right="12.678383" left="11.921057" top="0.519094" bottom="0.082900"/>
        <limitY/>
        <curve color="#1ac938" name="/iiwa_front/bspline_ff_joint_trajectory_controller/state/actual/positions.2"/>
        <curve color="#1f77b4" name="/iiwa_front/bspline_ff_joint_trajectory_controller/state/desired/positions.2"/>
       </plot>
      </DockArea>
     </DockSplitter>
     <DockSplitter count="4" orientation="|" sizes="0.25;0.25;0.25;0.25">
      <DockArea name="...">
       <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
        <range right="12.678383" left="11.921057" top="-0.351613" bottom="-0.598876"/>
        <limitY/>
        <curve color="#ff7f0e" name="/iiwa_front/bspline_ff_joint_trajectory_controller/state/actual/positions.3"/>
        <curve color="#d62728" name="/iiwa_front/bspline_ff_joint_trajectory_controller/state/desired/positions.3"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
        <range right="12.678383" left="11.921057" top="0.276650" bottom="0.036884"/>
        <limitY/>
        <curve color="#f14cc1" name="/iiwa_front/bspline_ff_joint_trajectory_controller/state/actual/positions.4"/>
        <curve color="#1ac938" name="/iiwa_front/bspline_ff_joint_trajectory_controller/state/desired/positions.4"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
        <range right="12.678383" left="11.921057" top="1.612641" bottom="0.751777"/>
        <limitY/>
        <curve color="#9467bd" name="/iiwa_front/bspline_ff_joint_trajectory_controller/state/actual/positions.5"/>
        <curve color="#ff7f0e" name="/iiwa_front/bspline_ff_joint_trajectory_controller/state/desired/positions.5"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
        <range right="12.678383" left="11.921057" top="-0.483001" bottom="-0.749011"/>
        <limitY/>
        <curve color="#1f77b4" name="/iiwa_front/bspline_ff_joint_trajectory_controller/state/actual/positions.6"/>
        <curve color="#d62728" name="/iiwa_front/bspline_ff_joint_trajectory_controller/state/desired/positions.6"/>
       </plot>
      </DockArea>
     </DockSplitter>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab containers="1" tab_name="vel">
   <Container>
    <DockSplitter count="2" orientation="-" sizes="0.500564;0.499436">
     <DockSplitter count="3" orientation="|" sizes="0.334192;0.331616;0.334192">
      <DockArea name="...">
       <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
        <range right="12.678383" left="11.921057" top="0.638721" bottom="-0.227602"/>
        <limitY/>
        <curve color="#f14cc1" name="/iiwa_front/bspline_ff_joint_trajectory_controller/state/desired/velocities.0"/>
        <curve color="#1ac938" name="/iiwa_front/bspline_ff_joint_trajectory_controller/state/actual/velocities.0"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
        <range right="12.678383" left="11.921057" top="1.292489" bottom="-1.135849"/>
        <limitY/>
        <curve color="#9467bd" name="/iiwa_front/bspline_ff_joint_trajectory_controller/state/desired/velocities.1"/>
        <curve color="#ff7f0e" name="/iiwa_front/bspline_ff_joint_trajectory_controller/state/actual/velocities.1"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
        <range right="12.678383" left="11.921057" top="1.565785" bottom="-1.782388"/>
        <limitY/>
        <curve color="#17becf" name="/iiwa_front/bspline_ff_joint_trajectory_controller/state/desired/velocities.2"/>
        <curve color="#f14cc1" name="/iiwa_front/bspline_ff_joint_trajectory_controller/state/actual/velocities.2"/>
       </plot>
      </DockArea>
     </DockSplitter>
     <DockSplitter count="3" orientation="|" sizes="0.334192;0.331616;0.334192">
      <DockArea name="...">
       <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
        <range right="12.678383" left="11.921057" top="1.118338" bottom="-0.661992"/>
        <limitY/>
        <curve color="#bcbd22" name="/iiwa_front/bspline_ff_joint_trajectory_controller/state/desired/velocities.3"/>
        <curve color="#9467bd" name="/iiwa_front/bspline_ff_joint_trajectory_controller/state/actual/velocities.3"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
        <range right="12.678383" left="11.921057" top="1.846690" bottom="-0.716694"/>
        <limitY/>
        <curve color="#1f77b4" name="/iiwa_front/bspline_ff_joint_trajectory_controller/state/desired/velocities.4"/>
        <curve color="#17becf" name="/iiwa_front/bspline_ff_joint_trajectory_controller/state/actual/velocities.4"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
        <range right="12.678383" left="11.921057" top="2.375415" bottom="-2.291205"/>
        <limitY/>
        <curve color="#d62728" name="/iiwa_front/bspline_ff_joint_trajectory_controller/state/desired/velocities.5"/>
        <curve color="#bcbd22" name="/iiwa_front/bspline_ff_joint_trajectory_controller/state/actual/velocities.5"/>
       </plot>
      </DockArea>
     </DockSplitter>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab containers="1" tab_name="acc">
   <Container>
    <DockSplitter count="2" orientation="-" sizes="0.500564;0.499436">
     <DockSplitter count="3" orientation="|" sizes="0.334192;0.331616;0.334192">
      <DockArea name="...">
       <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
        <range right="12.678383" left="11.921057" top="7.933772" bottom="-8.571606"/>
        <limitY/>
        <curve color="#1f77b4" name="/iiwa_front/bspline_ff_joint_trajectory_controller/state/desired/accelerations.0"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
        <range right="12.678383" left="11.921057" top="4.178326" bottom="-12.810809"/>
        <limitY/>
        <curve color="#d62728" name="/iiwa_front/bspline_ff_joint_trajectory_controller/state/desired/accelerations.1"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
        <range right="12.678383" left="11.921057" top="9.962018" bottom="-15.234764"/>
        <limitY/>
        <curve color="#1ac938" name="/iiwa_front/bspline_ff_joint_trajectory_controller/state/desired/accelerations.2"/>
       </plot>
      </DockArea>
     </DockSplitter>
     <DockSplitter count="3" orientation="|" sizes="0.334192;0.331616;0.334192">
      <DockArea name="...">
       <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
        <range right="12.678383" left="11.921057" top="11.522029" bottom="-10.984929"/>
        <limitY/>
        <curve color="#ff7f0e" name="/iiwa_front/bspline_ff_joint_trajectory_controller/state/desired/accelerations.3"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
        <range right="12.678383" left="11.921057" top="12.369834" bottom="-14.724373"/>
        <limitY/>
        <curve color="#f14cc1" name="/iiwa_front/bspline_ff_joint_trajectory_controller/state/desired/accelerations.4"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
        <range right="12.678383" left="11.921057" top="31.492444" bottom="-4.370065"/>
        <limitY/>
        <curve color="#9467bd" name="/iiwa_front/bspline_ff_joint_trajectory_controller/state/desired/accelerations.5"/>
       </plot>
      </DockArea>
     </DockSplitter>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab containers="1" tab_name="tab2">
   <Container>
    <DockSplitter count="1" orientation="-" sizes="1">
     <DockSplitter count="2" orientation="|" sizes="0.5;0.5">
      <DockArea name="...">
       <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
        <range right="12.678383" left="11.921057" top="0.973441" bottom="0.340604"/>
        <limitY/>
        <curve color="#d62728" name="/iiwa_front/joint_states/F_striker_joint_1/position"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
        <range right="12.678383" left="11.921057" top="0.104035" bottom="-0.001931"/>
        <limitY/>
        <curve color="#1ac938" name="/iiwa_front/joint_states/F_striker_joint_2/position"/>
       </plot>
      </DockArea>
     </DockSplitter>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab containers="1" tab_name="eff">
   <Container>
    <DockSplitter count="2" orientation="-" sizes="0.500564;0.499436">
     <DockSplitter count="3" orientation="|" sizes="0.333548;0.332904;0.333548">
      <DockArea name="...">
       <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
        <range right="12.678383" left="11.921057" top="15.921916" bottom="-45.213889"/>
        <limitY/>
        <curve color="#1f77b4" name="/iiwa_front/joint_states/F_joint_1/effort"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
        <range right="12.678383" left="11.921057" top="-40.861777" bottom="-105.807197"/>
        <limitY/>
        <curve color="#d62728" name="/iiwa_front/joint_states/F_joint_2/effort"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
        <range right="12.678383" left="11.921057" top="11.918136" bottom="-12.459533"/>
        <limitY/>
        <curve color="#1ac938" name="/iiwa_front/joint_states/F_joint_3/effort"/>
       </plot>
      </DockArea>
     </DockSplitter>
     <DockSplitter count="3" orientation="|" sizes="0.333548;0.332904;0.333548">
      <DockArea name="...">
       <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
        <range right="12.678383" left="11.921057" top="40.258246" bottom="12.208654"/>
        <limitY/>
        <curve color="#ff7f0e" name="/iiwa_front/joint_states/F_joint_4/effort"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
        <range right="12.678383" left="11.921057" top="9.134869" bottom="-7.148181"/>
        <limitY/>
        <curve color="#f14cc1" name="/iiwa_front/joint_states/F_joint_5/effort"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot style="Lines" mode="TimeSeries" flip_x="false" flip_y="false">
        <range right="12.678383" left="11.921057" top="2.863341" bottom="-8.022529"/>
        <limitY/>
        <curve color="#9467bd" name="/iiwa_front/joint_states/F_joint_6/effort"/>
       </plot>
      </DockArea>
     </DockSplitter>
    </DockSplitter>
   </Container>
  </Tab>
  <currentTabIndex index="0"/>
 </tabbed_widget>
 <use_relative_time_offset enabled="1"/>
 <!-- - - - - - - - - - - - - - - -->
 <!-- - - - - - - - - - - - - - - -->
 <Plugins>
  <plugin ID="DataLoad CSV">
   <default delimiter="0" time_axis=""/>
  </plugin>
  <plugin ID="DataLoad ROS bags">
   <use_header_stamp value="false"/>
   <discard_large_arrays value="true"/>
   <max_array_size value="100"/>
   <boolean_strings_to_number value="true"/>
   <remove_suffix_from_strings value="true"/>
  </plugin>
  <plugin ID="DataLoad ULog"/>
  <plugin ID="ROS Topic Subscriber">
   <use_header_stamp value="false"/>
   <discard_large_arrays value="true"/>
   <max_array_size value="100"/>
   <boolean_strings_to_number value="true"/>
   <remove_suffix_from_strings value="true"/>
  </plugin>
  <plugin ID="UDP Server"/>
  <plugin ID="WebSocket Server"/>
  <plugin ID="Fast Fourier Transform"/>
  <plugin ID="Quaternion to RPY"/>
  <plugin ID="Reactive Script Editor">
   <library code="--[[ Helper function to create a series from arrays&#xa;&#xa; new_series: a series previously created with ScatterXY.new(name)&#xa; prefix:     prefix of the timeseries, before the index of the array&#xa; suffix_X:   suffix to complete the name of the series containing the X value. If [nil], use the index of the array.&#xa; suffix_Y:   suffix to complete the name of the series containing the Y value&#xa; timestamp:   usually the tracker_time variable&#xa;              &#xa; Example:&#xa; &#xa; Assuming we have multiple series in the form:&#xa; &#xa;   /trajectory/node.{X}/position/x&#xa;   /trajectory/node.{X}/position/y&#xa;   &#xa; where {N} is the index of the array (integer). We can create a reactive series from the array with:&#xa; &#xa;   new_series = ScatterXY.new(&quot;my_trajectory&quot;) &#xa;   CreateSeriesFromArray( new_series, &quot;/trajectory/node&quot;, &quot;position/x&quot;, &quot;position/y&quot;, tracker_time );&#xa;]]--&#xa;&#xa;function CreateSeriesFromArray( new_series, prefix, suffix_X, suffix_Y, timestamp )&#xa;  &#xa;  --- clear previous values&#xa;  new_series:clear()&#xa;  &#xa;  --- Append points to new_series&#xa;  index = 0&#xa;  while(true) do&#xa;&#xa;    x = index;&#xa;    -- if not nil, get the X coordinate from a series&#xa;    if suffix_X ~= nil then &#xa;      series_x = TimeseriesView.find( string.format( &quot;%s.%d/%s&quot;, prefix, index, suffix_X) )&#xa;      if series_x == nil then break end&#xa;      x = series_x:atTime(timestamp)&#x9; &#xa;    end&#xa;    &#xa;    series_y = TimeseriesView.find( string.format( &quot;%s.%d/%s&quot;, prefix, index, suffix_Y) )&#xa;    if series_y == nil then break end &#xa;    y = series_y:atTime(timestamp)&#xa;    &#xa;    new_series:push_back(x,y)&#xa;    index = index+1&#xa;  end&#xa;end&#xa;&#xa;--[[ Similar to the built-in function GetSeriesNames(), but select only the names with a give prefix. ]]--&#xa;&#xa;function GetSeriesNamesByPrefix(prefix)&#xa;  -- GetSeriesNames(9 is a built-in function&#xa;  all_names = GetSeriesNames()&#xa;  filtered_names = {}&#xa;  for i, name in ipairs(all_names)  do&#xa;    -- check the prefix&#xa;    if name:find(prefix, 1, #prefix) then&#xa;      table.insert(filtered_names, name);&#xa;    end&#xa;  end&#xa;  return filtered_names&#xa;end&#xa;&#xa;--[[ Modify an existing series, applying offsets to all their X and Y values&#xa;&#xa; series: an existing timeseries, obtained with TimeseriesView.find(name)&#xa; delta_x: offset to apply to each x value&#xa; delta_y: offset to apply to each y value  &#xa;&#xa;]]--&#xa;&#xa;function ApplyOffsetInPlace(series, delta_x, delta_y)&#xa;  -- use C++ indeces, not Lua indeces&#xa;  for index=0, series:size()-1 do&#xa;    x,y = series:at(index)&#xa;    series:set(index, x + delta_x, y + delta_y)&#xa;  end&#xa;end&#xa;"/>
   <scripts/>
  </plugin>
  <plugin ID="CSV Exporter"/>
  <plugin ID="ROS /rosout Visualization"/>
  <plugin ID="ROS Topic Re-Publisher"/>
 </Plugins>
 <!-- - - - - - - - - - - - - - - -->
 <previouslyLoaded_Datafiles/>
 <previouslyLoaded_Streamer name="ROS Topic Subscriber"/>
 <!-- - - - - - - - - - - - - - - -->
 <customMathEquations/>
 <snippets/>
 <!-- - - - - - - - - - - - - - - -->
</root>

